dv/search_indexes.py

```diff
@@ -315,3 +314,146 @@ def prepare_geotarget(self, obj):
     def prepare_geotarget_auto(self, obj):
         geotargets = self.prepare_geotarget(obj)
         return ' '.join(geotargets) if geotargets else None
+
+
+class NewsIndex(indexes.SearchIndex, indexes.Indexable):
+    # common facets;
+    state_name = indexes.FacetMultiValueField()
+    financial_mechanism_ss = indexes.FacetMultiValueField()
+    programme_area_ss = indexes.FacetMultiValueField()
+    priority_sector_ss = indexes.FacetMultiValueField()
+    programme_name = indexes.FacetMultiValueField()
+    programme_status = indexes.FacetMultiValueField()
+    outcome_ss = indexes.FacetMultiValueField()
+
+    kind = indexes.FacetCharField()
+
+    # specific facets
+    project_name = indexes.FacetMultiValueField()
+    project_status = indexes.FacetMultiValueField()
+    geotarget = indexes.FacetCharField()
+    theme_ss = indexes.FacetMultiValueField()
+
+    # specific fields
+    text = indexes.CharField(document=True, use_template=True)
+    summary = indexes.CharField(model_attr='summary', indexed=False)
+
+    # extra data; avoid db hit
+    name = indexes.CharField(model_attr='title', indexed=False)
+    url = indexes.CharField(model_attr='link', indexed=False)
+    image = indexes.CharField(model_attr='image', indexed=False)
+
+    def get_model(self):
+        return News
+
+    def prepare_kind(self, obj):
+        return 'News'
+
+    def prepare_state_name(self, obj):
+        if obj.programmes.exists():
+            return list(obj.programmes.values_list('state__name', flat=True).distinct())
+        try:
+            if obj.project:
+                return [obj.project.state.name]
+        except Project.DoesNotExist:
+            pass
+        return None
+
+    def prepare_financial_mechanism_ss(self, obj):
+        try:
+            if obj.project:
+                return [obj.project.financial_mechanism.grant_name]
+        except Project.DoesNotExist:
```

Comment:
1. when will we get an exception?
2. what about news without project, but linked to a programme?